File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
from collections import deque
def solution(arr):
    answer = []
    
    for x in arr: #1,1,3,3,0,1,1
        if len(answer)==0:
            answer.append(x)
        else:
            k = answer[-1]
            if x == k:
                continue
            else:
                answer.append(x)
    
        
    return answer

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        
        
        
        ArrayList<Integer> lst = new ArrayList<>();
        
        for(int num : arr){
            if (lst.isEmpty() || lst.get(lst.size()-1) != num){
                lst.add(num);
            }
        }
        
        int[] answer = new int[lst.size()];
        for (int i=0; i<lst.size(); i++){
            answer[i] = lst.get(i); //i번째 인덱스 원소 가져오기
        }
        
        return answer;
    }
}